File: RxTools-library/src/main/java/com/vondear/rxtools/view/dialog/RxDialogScaleView.java
package com.vondear.rxtools.view.dialog;

import android.app.Activity;
import android.content.Context;
import android.graphics.Bitmap;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;

import com.vondear.rxtools.R;
import com.vondear.rxtools.view.scaleimage.ImageSource;
import com.vondear.rxtools.view.scaleimage.RxScaleImageView;


/**
 * @author vondear
 * @date 2016/7/19
 * 查看图片并支持手势缩放
 */
public class RxDialogScaleView extends RxDialog {

    private RxScaleImageView mRxScaleImageView;
    private String filePath;
    private Uri fileUri;
    private String fileAssetName;
    private Bitmap fileBitmap;
    private int resId;
    private int maxScale = 100;

    public RxDialogScaleView(Context context) {
        super(context);
        initView();
    }

    public RxDialogScaleView(Activity context) {
        super(context);
        initView();
    }


    public RxDialogScaleView(Context context,String filePath,boolean isAssets) {
        super(context);
        initView();
        setImage(filePath,isAssets);
    }

    public RxDialogScaleView(Context context,Uri uri) {
        super(context);
        initView();
        setImage(uri);
    }

    public RxDialogScaleView(Context context,int resId,boolean isResId) {
        super(context);
        initView();
        if (isResId) {
            setImage(resId);
        }
    }

    public RxDialogScaleView(Context context,Bitmap bitmap) {
        super(context);
        initView();
        setImage(bitmap);
    }

    public RxDialogScaleView(Context context, int themeResId) {
        super(context, themeResId);
        initView();
    }

    public RxDialogScaleView(Context context, boolean cancelable, OnCancelListener cancelListener) {
        super(context, cancelable, cancelListener);
        initView();
    }


    public RxDialogScaleView(Context context, float alpha, int gravity) {
        super(context, alpha, gravity);
        initView();
    }

    public RxScaleImageView getRxScaleImageView() {
        return mRxScaleImageView;
    }

    public void setImage(String filePath,boolean isAssets) {
        if (isAssets) {
            this.fileAssetName = fileAssetName;
            mRxScaleImageView.setImage(ImageSource.asset(filePath));
        }else {
            this.filePath = filePath;
            mRxScaleImageView.setImage(ImageSource.uri(filePath));
        }
    }

    public void setImage(Uri uri) {
        this.fileUri = uri;
        mRxScaleImageView.setImage(ImageSource.uri(uri));
    }

    public void setImage(int resId) {
        this.resId = resId;
        mRxScaleImageView.setImage(ImageSource.resource(resId));
    }

    public void setImage(Bitmap bitmap) {
        this.fileBitmap = bitmap;
        mRxScaleImageView.setImage(ImageSource.bitmap(fileBitmap));
    }

    private void initView() {
        View dialogView = LayoutInflater.from(mContext).inflate(R.layout.dialog_scaleview, null);
        mRxScaleImageView = dialogView.findViewById(R.id.rx_scale_view);
        mRxScaleImageView.setMaxScale(maxScale);
        ImageView ivClose =  dialogView.findViewById(R.id.iv_close);
        ivClose.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                cancel();
            }
        });
        setFullScreen();
        setContentView(dialogView);
    }

    public int getMaxScale() {
        return maxScale;
    }

    public void setMaxScale(int maxScale) {
        this.maxScale = maxScale;
        initView();
    }

    public String getFilePath() {
        return filePath;
    }

    public Uri getFileUri() {
        return fileUri;
    }

    public String getFileAssetName() {
        return fileAssetName;
    }

    public Bitmap getFileBitmap() {
        return fileBitmap;
    }

    public int getResId() {
        return resId;
    }


}

File: RxUI/src/main/java/com/tamsiree/rxui/view/dialog/RxDialogScaleView.kt
package com.tamsiree.rxui.view.dialog

import android.app.Activity
import android.content.Context
import android.content.DialogInterface
import android.graphics.Bitmap
import android.net.Uri
import android.view.LayoutInflater
import android.widget.ImageView
import com.tamsiree.rxui.R
import com.tamsiree.rxui.view.scaleimage.ImageSource
import com.tamsiree.rxui.view.scaleimage.RxScaleImageView

/**
 * @author tamsiree
 * @date 2016/7/19
 * 查看图片并支持手势缩放
 */
class RxDialogScaleView : RxDialog {
    lateinit var rxScaleImageView: RxScaleImageView
        private set
    var filePath: String? = null
        private set
    var fileUri: Uri? = null
        private set
    var fileAssetName: String? = null
        private set
    var fileBitmap: Bitmap? = null
        private set
    var resId = 0
        private set
    private var maxScale = 100

    constructor(context: Context?) : super(context!!) {
        initView()
    }

    constructor(context: Activity?) : super(context!!) {
        initView()
    }

    constructor(context: Context?, filePath: String?, isAssets: Boolean) : super(context!!) {
        initView()
        setImage(filePath, isAssets)
    }

    constructor(context: Context?, uri: Uri?) : super(context!!) {
        initView()
        setImage(uri)
    }

    constructor(context: Context?, resId: Int, isResId: Boolean) : super(context!!) {
        initView()
        if (isResId) {
            setImage(resId)
        }
    }

    constructor(context: Context?, bitmap: Bitmap?) : super(context!!) {
        initView()
        setImage(bitmap)
    }

    constructor(context: Context?, themeResId: Int) : super(context!!, themeResId) {
        initView()
    }

    constructor(context: Context?, cancelable: Boolean, cancelListener: DialogInterface.OnCancelListener?) : super(context!!, cancelable, cancelListener) {
        initView()
    }

    constructor(context: Context?, alpha: Float, gravity: Int) : super(context, alpha, gravity) {
        initView()
    }

    fun setImage(filePath: String?, isAssets: Boolean) {
        if (isAssets) {
            fileAssetName = fileAssetName
            rxScaleImageView.setImage(ImageSource.asset(filePath))
        } else {
            this.filePath = filePath
            rxScaleImageView.setImage(ImageSource.uri(filePath))
        }
    }

    fun setImage(uri: Uri?) {
        fileUri = uri
        rxScaleImageView.setImage(ImageSource.uri(uri))
    }

    fun setImage(resId: Int) {
        this.resId = resId
        rxScaleImageView.setImage(ImageSource.newImageResource(resId))
    }

    fun setImage(bitmap: Bitmap?) {
        fileBitmap = bitmap
        rxScaleImageView.setImage(ImageSource.bitmap(fileBitmap))
    }

    private fun initView() {
        val dialogView = LayoutInflater.from(mContext).inflate(R.layout.dialog_scaleview, null)
        rxScaleImageView = dialogView.findViewById(R.id.rx_scale_view)
        rxScaleImageView.maxScale = maxScale.toFloat()
        val ivClose = dialogView.findViewById<ImageView>(R.id.iv_close)
        ivClose.setOnClickListener { cancel() }
        setFullScreen()
        setContentView(dialogView)
    }

    fun getMaxScale(): Int {
        return maxScale
    }

    fun setMaxScale(maxScale: Int) {
        this.maxScale = maxScale
        initView()
    }

}